swap/src/database.rs

```diff
@@ -110,6 +116,46 @@ impl Database {
         Ok(PeerId::from_str(peer_id.as_str())?)
     }
 
+    pub async fn insert_address(&self, peer_id: PeerId, address: Multiaddr) -> Result<()> {
```

Comment:
Did you test this? Could we have a database test that writes and reads addresses?

Reply:
I tested it by doing a swap and resuming it. Will see to add a test!

Reply:
if you can please don't rebase - I based #593 on this